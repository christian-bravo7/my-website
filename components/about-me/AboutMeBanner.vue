<template>
  <AppBanner>
    <template #primary-text>
      <span class="block text-center">
        {{ $t('me.simple-description') }}
      </span>
    </template>
    <template #secondary-text>
      <span class="block text-center">
        {{ $t('me.about-label') }}
      </span>
      <span class="block text-center">
        {{ $t('me.me-label') }}
      </span>
    </template>
  </AppBanner>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class AboutMeBanner extends Vue {}
</script>
