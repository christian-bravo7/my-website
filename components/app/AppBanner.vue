<template>
  <transition
    name="flash-text"
    @enter="shuffleText"
  >
    <section class="flex justify-center items-center px-4">
      <div class="relative">
        <h2
          id="secondary"
          class="absolute font-bold text-center text-8xl sm:text-10xl md:text-12xl leading-none select-none text-gray-200 dark:text-blue-700 secondary-text"
        >
          <slot name="secondary-text" />
        </h2>
        <h1
          id="primary"
          class="relative max-w-screen-md text-center text-3xl md:text-5xl z-10 text-gray-900 dark:text-gray-50 primary-text"
        >
          <slot name="primary-text" />
        </h1>
      </div>
    </section>
  </transition>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import baffle from 'baffle';

@Component
export default class AppBanner extends Vue {
  shuffleText (): void {
    // @ts-ignore
    const secondaryText = baffle('#secondary', { speed: 80 });
    secondaryText.reveal(1000);
  }
}

</script>

<style lang="scss" scoped>

.primary-text {
  transition-duration: 2000ms;
  transition-property: opacity;
}

.secondary-text {
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.flash-text-enter {
  .primary-text {
    opacity: 0;
  }
}

.flash-text-enter-active {
  transition: 5s;
}

.flash-text-enter-to {
  .primary-text {
    opacity: 1;
  }
}
</style>
