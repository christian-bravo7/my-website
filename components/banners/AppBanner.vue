<template>
  <Transition
    name="flash-text"
    @enter="shuffleText"
    @after-enter="displayPrimaryText"
  >
    <section class="relative | flex justify-center items-center | px-4">
      <div class="relative">
        <Transition name="secondary-text">
          <h2
            v-show="isSecondaryTextVisible"
            id="secondary"
            class="absolute | font-bold text-center | text-6xl sm:text-8xl md:text-10xl xl:text-13xl | leading-none select-none | text-gray-100 dark:text-blue-900 | app-banner__secondary-text"
          >
            <slot name="secondary-text" />
          </h2>
        </Transition>
        <Transition name="primary-text">
          <h1
            v-show="isPrimaryTextVisible"
            class="relative z-10 | max-w-md sm:max-w-lg lg:max-w-screen-md xl:max-w-screen-lg | text-center text-xl sm:text-3xl md:text-4xl xl:text-5xl | text-gray-900 dark:text-gray-50 | app-banner__primary-text"
          >
            <slot name="primary-text" />
          </h1>
        </Transition>
      </div>
    </section>
  </Transition>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import baffle from 'baffle';

@Component
export default class AppBanner extends Vue {
  isPrimaryTextVisible: boolean = false;
  isSecondaryTextVisible: boolean = false;

  shuffleText (): void {
    this.displaySecondaryText();
    // @ts-ignore
    const secondaryText = baffle('#secondary', { speed: 70 });
    secondaryText.reveal(1000);
  }

  displayPrimaryText (): void {
    this.isPrimaryTextVisible = true;
  }

  displaySecondaryText (): void {
    this.isSecondaryTextVisible = true;
  }
}

</script>

<style lang="scss" scoped>
.app-banner {
  &__secondary-text {
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
}

.flash-text-enter-active {
  transition: 0.9s;
}

.secondary-text-enter {
  opacity: 0;
}

.secondary-text-enter-to {
  opacity: 1;
}

.secondary-text-enter-active {
  transition-duration: 0.8s;
  transition-property: opacity;
}

.primary-text-enter {
  transform: translateY(50px);
  opacity: 0;
}

.primary-text-enter-to {
  transform: translateY(0);
  opacity: 1;
}

.primary-text-enter-active {
  transition-duration: 1s;
  transition-property: opacity, transform;
}
</style>
