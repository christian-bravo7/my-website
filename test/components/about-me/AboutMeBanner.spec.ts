import { createLocalVue, mount } from '@vue/test-utils';

import AboutMeBanner from '@/components/about-me/AboutMeBanner.vue';
import AppBanner from '@/components/app/AppBanner.vue';

import setupI18n from '@/test/config/i18n';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

const componentConfig = {
  localVue,
  i18n,
};

describe('AboutMeBanner component', () => {
  it('should render correctly', () => {
    const wrapper = mount(AboutMeBanner, {
      components: {
        AppBanner,
      },
      ...componentConfig,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render primary and seconday text', () => {
    const wrapper = mount(AboutMeBanner, {
      components: {
        AppBanner,
      },
      ...componentConfig,
    });

    const primaryText = wrapper.find('.app-banner__primary-text');
    const secondaryText = wrapper.find('#secondary');

    expect(primaryText.text()).toBe('Helping companies achieve their goals, doing what I love');
    expect(secondaryText.text()).toBe('About me');
  });
});
