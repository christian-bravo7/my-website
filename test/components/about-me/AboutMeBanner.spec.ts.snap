// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMeBanner component should render correctly 1`] = `
<section class="flex justify-center items-center | px-4">
  <div class="relative">
    <h2 id="secondary" class="absolute | font-bold text-center | text-6xl sm:text-8xl md:text-13xl | leading-none select-none | text-gray-200 dark:text-blue-900 | app-banner__secondary-text">
      navigation.about-me-label
    </h2>
    <h1 class="relative z-10 | max-w-xs sm:max-w-lg md:max-w-screen-md | text-center text-xl sm:text-3xl md:text-5xl | text-gray-900 dark:text-gray-50 | app-banner__primary-text" style="display: none;"><span class="block | text-center | about-me-banner__description">me.simple-description</span></h1>
  </div>
</section>
`;
